README
NAME

    Dist::Zilla::PluginBundle::Git::CheckFor - All Git::CheckFor plugins at
    once

VERSION

    This document describes version 0.013 of
    Dist::Zilla::PluginBundle::Git::CheckFor - released March 10, 2015 as
    part of Dist-Zilla-PluginBundle-Git-CheckFor.

SYNOPSIS

        ; in dist.ini
        [@Git::CheckFor]

DESCRIPTION

    This bundles several plugins that do some sanity/lint checking of your
    git repository; namely: you're on the right branch and you haven't
    forgotten any autosquash commits (fixup! or squash!).

SEE ALSO

    Please see those modules/websites for more information related to this
    module.

      * Dist::Zilla::Plugin::Git::CheckFor::Fixups

      * Dist::Zilla::Plugin::Git::CheckFor::CorrectBranch

      * Dist::Zilla::Plugin::Git::CheckFor::MergeConflicts

      * Dist::Zilla::PluginBundle::Git

SOURCE

    The development version is on github at
    http://https://github.com/RsrchBoy/dist-zilla-pluginbundle-git-checkfor
    and may be cloned from
    git://https://github.com/RsrchBoy/dist-zilla-pluginbundle-git-checkfor.git

BUGS

    Please report any bugs or feature requests on the bugtracker website
    https://github.com/RsrchBoy/dist-zilla-pluginbundle-git-checkfor/issues

    When submitting a bug or request, please include a test-file or a patch
    to an existing test-file that illustrates the bug or desired feature.

AUTHOR

    Chris Weyl <cweyl@alumni.drew.edu>

 I'm a material boy in a material world

    Please note I do not expect to be gittip'ed or flattr'ed for this work,
    rather it is simply a very pleasant surprise. I largely create and
    release works like this because I need them or I find it enjoyable;
    however, don't let that stop you if you feel like it ;)

    Flattr this
    <https://flattr.com/submit/auto?user_id=RsrchBoy&url=https%3A%2F%2Fgithub.com%2FRsrchBoy%2Fdist-zilla-pluginbundle-git-checkfor&title=RsrchBoy's%20CPAN%20Dist-Zilla-PluginBundle-Git-CheckFor&tags=%22RsrchBoy's%20Dist-Zilla-PluginBundle-Git-CheckFor%20in%20the%20CPAN%22>,
    gittip me <https://www.gittip.com/RsrchBoy/>, or indulge my Amazon
    Wishlist <http://bit.ly/rsrchboys-wishlist>... If you so desire.

CONTRIBUTORS

      * Christian Walde <walde.christian@googlemail.com>

      * Karen Etheridge <ether@cpan.org>

      * Mike Doherty <doherty@cs.dal.ca>

      * Olivier Mengué <dolmen@cpan.org>

COPYRIGHT AND LICENSE

    This software is Copyright (c) 2012 by Chris Weyl.

    This is free software, licensed under:

      The GNU Lesser General Public License, Version 2.1, February 1999



Changes
Revision history for Dist-Zilla-PluginBundle-Git-CheckFor

{{$NEXT}}
        * fix issue #26: . is no longer in @INC
        * remove dependency on MooseX::AttributeShortcuts, to work around
          installation issues on blead

0.014     2016-10-10 10:27:02 CDT-0500
	* Drop dependency on List::MoreUtils (thanks, @karenetheridge!)

0.013     2015-03-10 19:33:10 PDT-0700
	* Drop autobox usage (thanks, @karenetheridge!)

0.012     2015-02-23 20:47:40 PST-0800
	* Switch tests to Test::TempDir::Tiny (thanks @karenetheridge! closed #19)

0.011     2014-04-11 12:57:35 PDT-0700
	* Set GIT_AUTHOR_EMAIL, etc, in our tests to prevent git identity crises;
	  closes #13
	* Rework a commit message in t/plugin/fixups.t; this closes #16 (or
	  should, at any rate)

0.010     2014-04-11 11:48:23 PDT-0700
    * use CheckForBin, not RequiresExternal in our dzil config; this should
	  stop false fails from being reported (and be nicer in general) when git
	  is not installed on some system

0.009     2014-01-28 15:50:28 PST-0800
    * merge github #11, github #15

0.008     2013-11-10 22:55:17 PST-0800
    * fix issue with parsing utf8-encoded files

0.007     2013-09-29 17:04:08 PDT-0700
	* Fix issues with travis and our test suite (aka set user.name, user.email
	  in our test repositories)

0.006     2012-11-14 22:14:15 PST8PDT
	* Add plugin to check for common merge mishaps [Mike Doherty]
	* Add explicit dependency on the git executable

0.005     2012-10-03 22:10:53 PST8PDT
	* make our test suite usable on !unix platforms (github #4, #5)
	* Fix breakage introduced by the git NextVersion plugin in  1.122770 by
	  pulling over the removed code and integrating it here.  This is expected
	  to be a temporary measure as we figure out a more permanent (and
	  hopefully cooperative) solution.

0.004     2012-05-04 14:53:44 America/Los_Angeles
	* [Git::CheckFor::CorrectBranch] now permits releases from multiple brnaches
	  (Karen Etheridge)

0.003     2012-05-02 11:11:39 America/Los_Angeles
	* handle version tags where '-TRIAL' is appeneded.  We'll need to revisit this
	  later to make things Work Better, but for now it works and with a minimum of
	  fuss.

0.002     2012-04-13 14:16:08 America/Los_Angeles
	* no functional changes!
	* bump Test::Moose::More prereq to 0.008, as we need a version of
	  has_attribute_ok that handles checking roles w/o barfing (as Test::Moose now
	  seems to)

0.001     2012-02-06 13:11:50 America/Los_Angeles


t/funcs.pm
use strict;
use warnings;

use autodie 'system';
use IPC::System::Simple (); # for autodie && prereqs

use File::chdir;
use Capture::Tiny 'capture_merged';
use Try::Tiny;
use Path::Tiny;
use Test::TempDir::Tiny;

use Test::DZil;

$ENV{GIT_AUTHOR_EMAIL}    ||= 'Test Ing <test@test.ing>';
$ENV{GIT_COMMITTER_EMAIL} ||= 'Test Ing <test@test.ing>';

sub make_test_repo {
    my @commands = @_;

    my $tempdir = tempdir;
    my $repo_root = path($tempdir)->absolute;
    note "Repo being created at $repo_root";
    local $CWD = "$repo_root";

    unshift @commands,
        'git init',
        'git config user.name "No One"',
        'git config user.email "no@one.com"'
        unless $commands[0] =~ /git init/;

    # If we're not a code ref, make us one.
    @commands =
        map { my $x = $_; ref $x ? $x : sub { system "($x)" } }
        @commands
        ;

    # this is just to keep things quiet...
    my $fail = q{};
    my $merged = capture_merged {
        try { $_->() } catch { $fail .= $_ }
            for @commands;
    };

    die "Building git repo FAILED: $fail\n\nLOG:\n$merged"
        if $fail ne q{};

    return $repo_root;
}

sub _ack {
    my ($fn, $text, $msg) = @_;
    $text ||= 'whee';
    $msg  ||= 'ack';
    my $q = $^O eq 'MSWin32' ? '"' : "'";

    return (
        qq{echo "$text" >> $fn},
        qq{git add $fn && git commit -m $q$msg$q},
    );
}


sub prep_for_testing {
    # repo_init, plugin_list
    my %opts = @_;

    # make some basic commits, branch...
    my ($repo_root) = make_test_repo(@{$opts{repo_init}});

    my @dist_ini = @{$opts{plugin_list}};
    unshift @dist_ini, $opts{core_args} if $opts{core_args};

    # ...then create a Builder and check for exception
    my $tzil = Builder->from_config(
        { dist_root => "$repo_root" },
        {
            add_files => {
                'source/dist.ini' => simple_ini(@dist_ini),
            },
        },
    );

    return ($tzil, $repo_root);
}

# blatantly stolen from Dist-Zilla-Plugin-CheckPrereqsIndexed-0.008/t/basic.t
# Write the log messages as diagnostics:
sub diag_log
{
  my $tzil = shift;

  # Output nothing if all tests passed:
  my $all_passed = shift;
  $all_passed &&= $_ for @_;

  return if $all_passed;

  diag(map { "$_\n" } @{ $tzil->log_messages });
}

!!42;


Makefile.PL
#
# This file is part of Dist-Zilla-PluginBundle-Git-CheckFor
#
# This software is Copyright (c) 2012 by Chris Weyl.
#
# This is free software, licensed under:
#
#   The GNU Lesser General Public License, Version 2.1, February 1999
#
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v5.032.
use strict;
use warnings;

# inserted by Dist::Zilla::Plugin::CheckBin 0.007
use Devel::CheckBin;

use 5.006;

use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "All Git::CheckFor plugins at once",
  "AUTHOR" => "Chris Weyl <cweyl\@alumni.drew.edu>",
  "CONFIGURE_REQUIRES" => {
    "Devel::CheckBin" => 0,
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "Dist-Zilla-PluginBundle-Git-CheckFor",
  "EXE_FILES" => [],
  "LICENSE" => "lgpl",
  "MIN_PERL_VERSION" => "5.006",
  "NAME" => "Dist::Zilla::PluginBundle::Git::CheckFor",
  "PREREQ_PM" => {
    "Dist::Zilla::Plugin::Git::NextVersion" => "1.120370",
    "Dist::Zilla::Role::BeforeRelease" => 0,
    "Dist::Zilla::Role::Git::Repo" => 0,
    "Dist::Zilla::Role::PluginBundle::Easy" => 0,
    "Git::Wrapper" => 0,
    "IPC::System::Simple" => 0,
    "List::MoreUtils" => 0,
    "List::Util" => 0,
    "Moose" => 0,
    "Moose::Role" => 0,
    "MooseX::AttributeShortcuts" => 0,
    "Try::Tiny" => 0,
    "autodie" => 0,
    "namespace::autoclean" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "Capture::Tiny" => 0,
    "File::Spec" => 0,
    "File::chdir" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Path::Tiny" => 0,
    "Test::CheckDeps" => "0.010",
    "Test::DZil" => 0,
    "Test::Fatal" => 0,
    "Test::Moose::More" => "0.008",
    "Test::More" => "0.94",
    "Test::TempDir::Tiny" => 0
  },
  "VERSION" => "0.013",
  "test" => {
    "TESTS" => "t/*.t t/plugin/*.t"
  }
);


my %FallbackPrereqs = (
  "Capture::Tiny" => 0,
  "Devel::CheckBin" => 0,
  "Dist::Zilla::Plugin::Git::NextVersion" => "1.120370",
  "Dist::Zilla::Role::BeforeRelease" => 0,
  "Dist::Zilla::Role::Git::Repo" => 0,
  "Dist::Zilla::Role::PluginBundle::Easy" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "File::chdir" => 0,
  "Git::Wrapper" => 0,
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "IPC::System::Simple" => 0,
  "List::MoreUtils" => 0,
  "List::Util" => 0,
  "Moose" => 0,
  "Moose::Role" => 0,
  "MooseX::AttributeShortcuts" => 0,
  "Path::Tiny" => 0,
  "Test::CheckDeps" => "0.010",
  "Test::DZil" => 0,
  "Test::Fatal" => 0,
  "Test::Moose::More" => "0.008",
  "Test::More" => "0.94",
  "Test::TempDir::Tiny" => 0,
  "Try::Tiny" => 0,
  "autodie" => 0,
  "namespace::autoclean" => 0,
  "strict" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);
